core/src/main/java/org/apache/iceberg/MergingSnapshotUpdate.java

```diff
@@ -355,80 +351,110 @@ private ManifestFile filterManifest(Expression deleteExpression,
-        if (fileDelete || inclusive.eval(file.partition())) {
-          ValidationException.check(
-              fileDelete || strict.eval(file.partition()) || metricsEvaluator.eval(file),
-              "Cannot delete file where some, but not all, rows match filter %s: %s",
-              this.deleteExpression, file.path());
-
-          hasDeletedFiles = true;
-          if (failAnyDelete) {
-            throw new DeleteException(writeSpec().partitionToPath(file.partition()));
-          }
-          break; // as soon as a deleted file is detected, stop scanning
-        }
-      }
+      boolean hasDeletedFiles =
+          manifestHasDeletedFiles(metricsEvaluator, reader, inclusive, strict, pathWrapper, partitionWrapper);
 
       if (!hasDeletedFiles) {
         filteredManifests.put(manifest, manifest);
         return manifest;
       }
 
-      // when this point is reached, there is at least one file that will be deleted in the
-      // manifest. produce a copy of the manifest with all deleted files removed.
-      List<DataFile> deletedFiles = Lists.newArrayList();
-      Set<CharSequenceWrapper> deletedPaths = Sets.newHashSet();
-      OutputFile filteredCopy = manifestPath(manifestCount.getAndIncrement());
-      ManifestWriter writer = new ManifestWriter(reader.spec(), filteredCopy, snapshotId());
-      try {
-        for (ManifestEntry entry : reader.entries()) {
-          DataFile file = entry.file();
-          boolean fileDelete = (deletePaths.contains(pathWrapper.set(file.path())) ||
-              dropPartitions.contains(partitionWrapper.set(file.partition())));
-          if (entry.status() != Status.DELETED) {
-            if (fileDelete || inclusive.eval(file.partition())) {
-              ValidationException.check(
-                  fileDelete || strict.eval(file.partition()) || metricsEvaluator.eval(file),
-                  "Cannot delete file where some, but not all, rows match filter %s: %s",
-                  this.deleteExpression, file.path());
-
-              writer.delete(entry);
-
-              CharSequenceWrapper wrapper = CharSequenceWrapper.wrap(entry.file().path());
-              if (deletedPaths.contains(wrapper)) {
-                LOG.warn("Deleting a duplicate path from manifest {}: {}",
-                    manifest.path(), wrapper.get());
-                summaryBuilder.incrementDuplicateDeletes();
-              } else {
-                // only add the file to deletes if it is a new delete
-                // this keeps the snapshot summary accurate for non-duplicate data
-                deletedFiles.add(entry.file().copy());
-              }
-              deletedPaths.add(wrapper);
+      return filterManifestsWithDeletedFiles(
+          metricsEvaluator,
+          manifest,
+          reader,
+          inclusive,
+          strict,
+          pathWrapper,
+          partitionWrapper);
+    }
+  }
 
+  private boolean manifestHasDeletedFiles(
```

Comment:
I thought the convention we agreed on was multiple arguments per line until they no longer fit.